ci/test/Main.purs

```
@@ -160,6 +160,8 @@ goodPackageName = do
 
   parseName "a" "a"
   parseName "some-dash" "some-dash"
+  -- A blessed prefixed package
+  parseName "purescript-compiler-backend-utilities" "purescript-compiler-backend-utilities"
```

I am not sure if this is intentional, but we're changing the logic here: the current one will just strip the purescript prefix, but the new one will fail if there's any purescript prefix.
Unless we're doing the prefix-stripping somewhere else, this will fail whenever we try to parse anything into a `PackageName`.

---

We do the prefix-stripping elsewhere. This is a good thing because failing in `PackageName.parse` will tell us right away we forgot to strip a package name, versus silently succeeding and manipulating the package name along the way.

For example, we might shuffle code around and forget to verify there is no PureScript prefix when parsing package manifests; our code will process fine and the resulting package name will have no prefix in the registry, but its manifest file will still have the prefix. Instead, it would be better to fail right away so we can remember to update the code and verify the prefix is not present properly.

---

Sounds good!